### PR TITLE
Antag ticket/token system!

### PR DESCRIPTION
**Pull request by ghost**
:cl:
add: Antag tokens are an actual mechanic now!
admin: The antag token panel and redeem antag token buttons are in the Player Panel
/:cl:

### CODE REVIEW

**Comment by DaydreamPuppeh** [comment]
Remember, blob cannot be chosen. Tokens can't be used for team antags or during wiz/team antag rounds. Tokens can only be used in the first 5 minutes of a round. Also make a button for admins to disable tokens for a round. And make it so admins can accept or deny a token use if they are on.

**Comment by ghost** [comment]
This doesn't let you pick the gamemode. It only lets you use a token during the readying up stage, and using one gives you priority in whatever the mode is.

**Comment by k-k-karma** [comment]
ah so it's completely fucking useless and everyone will just wait fro an admin to get on

**Comment by ghost** [comment]
good news: admins can also manually redeem tokens

**Comment by k-k-karma** [comment]
What did I just say?

**Comment by ghost** [comment]
Should it be able to pick the gamemodes from a whitelist?

**Comment by k-k-karma** [comment]
please

**Comment by DaydreamPuppeh** [comment]
Blob shouldn't be allowed just because it's a bad mode on hippie.

**Comment by ghost** [comment]
so which modes SHOULD be allowed?

**Comment by DaydreamPuppeh** [comment]
Wiz, devil, hivemind(what a waste), lone op, lone clown op, infiltrator(what a waste), vampire, traitor, and wizard. Might have forgotten a few

**Comment by ghost** [comment]
devil is also a waste btw

**Comment by cris5311** [comment]
-1

The system is good as it is and this is bad. Before you could do cool stuff, but now its just you either get tator or you know its a team antag round. This is quite redundant and might acctually make the system worse as some of the not so good admins (the minority) will tell you to use the system when you want to redeem your token, which is already a very scarce good. 
If you gave everyone a toke for, lets say every 30 hours played, this would be good, but since tokens can (for normal players at least) only be obtained by admin intervention it is only logical that they can be used by admin intervention.

**Comment by ghost** [comment]
> If you gave everyone a toke for, lets say every 30 hours played, this would be good, but since tokens can (for normal players at least)

that's out of scope, though

**Comment by DaydreamPuppeh** [comment]
Anyway, +1

**Comment by ghost** [comment]
also, reminder, this requires both a SQL and config update!

**Comment by Manly-Dwarf** [comment]
Do not let antag tokens pick gamemodes, or rig rounds.

**Comment by ghost** [comment]
ok. good that cuts out alot of snowflake code.

**Comment by ghost** [comment]
ok, gamemode forcing/rigging removed.

**Comment by MaterialisticThings** [comment]
+1 2 THANOS' IS EPIC

**Comment by AnthonyJNgo** [comment]
+1 but I want batty reviewing this.

**Review by yoyobatty — DISMISSED** [review]
A lot of code I really don't see anything wrong with, will need updates to the database

**Comment by ghost** [comment]
@steamp0rt resolve these conflicts manually on your end, it shouldn't take too long.

File: code/game/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -1,590 +1,596 @@
...
-
-/datum/game_mode/proc/get_players_for_role(role)
-	var/list/players = list()
-	var/list/candidates = list()
-	var/list/drafted = list()
-	var/datum/mind/applicant = null
-
-	// Ultimate randomizing code right here
-	for(var/mob/dead/new_player/player in GLOB.player_list)
+
```
Comment (ghost) [inline code/game/gamemodes/game_mode.dm:1]: stupid newlines

File: hippiestation.dme [hunk 2]
```diff
@@ -1,3242 +1,3247 @@
...
-#include "code\__DEFINES\configuration.dm"
-#include "code\__DEFINES\construction.dm"
-#include "code\__DEFINES\contracts.dm"
-#include "code\__DEFINES\cult.dm"
-#include "code\__DEFINES\diseases.dm"
-#include "code\__DEFINES\DNA.dm"
-#include "code\__DEFINES\economy.dm"
-#include "code\__DEFINES\events.dm"
-#include "code\__DEFINES\exports.dm"
+// DM Environment file for tgstation.dme.
```
Comment (ghost) [inline hippiestation.dme]: stupid newlines

File: hippiestation/SQL/antag_tickets.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE `antag_tickets` (
```
Comment (ghost) [inline hippiestation/SQL/antag_tickets.sql:1]: this needs to be added to the SQL after merging!!

**Comment by Manly-Dwarf** [comment]
can we merge this already

**Comment by ghost** [comment]
lmao conflicts

**Comment by lunaman22** [comment]
+1 naenae

**Comment by ghost** [comment]
based

**Comment by Manly-Dwarf** [comment]
please fix this so it can be merged @steamp0rt

**Comment by Manly-Dwarf** [comment]
nice

**Comment by Majick1** [comment]
-1 honestly, with all the restrictions on it I have to agree with Cris, only allowing it on solo antag types doesn't really add much. Also tokens aren't common enough to not be able to have an admin manually do it.

**Comment by ghost** [comment]
i forgot why this was a good idea

**Comment by ghost** [comment]
also i coded it while sleep deprived

**Comment by Manly-Dwarf** [comment]
>closed

bruh moment

**Comment by theworldo** [comment]
hah

**Pull request closed without merging**